### PR TITLE
Op registration: linalg

### PR DESCRIPTION
**Pull request by asi1024**
Part of #6692.
Depends on #6820.

### CODE REVIEW

File: chainerx_cc/chainerx/routines/linalg.h [hunk 1]
```diff
@@ -4,9 +4,21 @@
...
+public:
+    static const char* name() { return "Dot"; }
+
+    virtual void Call(const Array& a, const Array& b, const Array& out) = 0;
+};
+
+// Matrix multiplication. All the operands are matrices (i.e., two-dimensional arrays).
+// Let the shapes of `a` and `b` be `(M, K)` and `(L, N)`, respectively.
+// Then, it must hold that `K == L` and the shape of `out` must be `(M, N)`.
+// Otherwise, the behavior is undefined.
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/linalg.h]: This is not the correct location for the comment. Put it to `DotOp`.

Comment (asi1024) [reply chainerx_cc/chainerx/routines/linalg.h]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test  this please

**Comment by niboshi** [comment]
LGTM

**Comment by hvy** [comment]
Removing `to-be-backported` as we decided not to backport non-critical PRs.

**Pull request merged**